src/keysym.c

```diff
@@ -296,26 +298,14 @@ xkb_keysym_from_name(const char *name, enum xkb_keysym_flags flags)
             }
         }
         if (entry) {
-            const struct name_keysym *iter, *last;
-
-            if (icase && xkb_keysym_is_lower(entry->keysym))
```

Comment:
Can we keep the `icase` check? I know that we already have a fast path for `!icase` above so `icase` is guaranteed here, but it's nice to keep respecting it in the slow-path for possible comparison/debugging purposes.

Reply:
I gave it a try, but I am really not convinced it helps: I know later I will wonder _why_ is this test there! I could add a comment, but then we already have `Find the correct keysym for case-insensitive match.` heading the slow-path comment block.

Reply:
OK

Reply:
TO be sure: Do you want to keep the new test (and squash) or should I drop the corresponding commit?

Reply:
The fix is quite important, so I am going to merge _without_ the extra test, but I am open to add it in a further PR.